python/lsst/daf/persistence/swiftStorage.py

```diff
@@ -0,0 +1,555 @@
+            Tuple, the first item is the path without the fits header, the
+            second item is the part that was stripped, if any.
+        """
+        strippedPath = path
+        pathStripped = ''
+        firstBracket = path.find("[")
+        if firstBracket != -1:
+            strippedPath = path[:firstBracket]
+            pathStripped = path[firstBracket:]
+        return strippedPath, pathStripped
+
+    def instanceSearch(self, path):
+        """Search for the given path in this storage instance.
+
+        If the path contains an HDU indicator (a number in brackets before the
+        dot, e.g. 'foo.fits[1]', this will be stripped when searching and so
+        will match filenames without the HDU indicator, e.g. 'foo.fits'. The
+        path returned WILL contain the indicator though, e.g. ['foo.fits[1]'].
+
+        Parameters
+        ----------
+        path : string
+            A filename (and optionally prefix path) to search for within root.
+
+        Returns
+        -------
+        string or None
+            The location that was found, or None if no location was found.
+        """
+        # Now search for the path in the root or its parents
+        # Strip off any cfitsio bracketed extension if present
+        strippedPath, pathStripped = self.getFitsHeaderStrippedPath(path)
+        try:
+            headers, objects = self._connection.get_container(
+                self._containerName)
+            locations = [obj['name'] for obj in objects]
+        except self.swift.ClientException as err:
+            raise RuntimeError("Container \'{0}\' not found: {1}".format(
+                self._containerName, err))
+        locations = fnmatch.filter(locations, strippedPath)
+        locations = [location + pathStripped for location in locations]
+        return locations if locations else None
+
+    @staticmethod
+    def search(uri, path):
+        """Look for the given path in the current root.
+
+        Also supports searching for the path in Butler v1 repositories by
+        following the Butler v1 _parent symlink
+
+        If the path contains an HDU indicator (a number in brackets, e.g.
+        'foo.fits[1]', this will be stripped when searching and so
+        will match filenames without the HDU indicator, e.g. 'foo.fits'. The
+        path returned WILL contain the indicator though, e.g. ['foo.fits[1]'].
+
+        Parameters
+        ----------
+        uri : string
+            The uri to the repository.
+        path : string
+            The path to the file within the root directory.
+
+        Returns
+        -------
+        string or None
+            The location that was found, or None if no location was found.
+        """
+        storage = SwiftStorage(uri, create=False)
+        return storage.instanceSearch(path) if storage else None
+
+    def copyFile(self, fromLocation, toLocation):
+        """Copy a file from one location to another on the local filesystem.
```

Comment:
From the implementation of this method I would say that the purpose here is to copy one Swift object into another Swift object, therefore in the same container, not in the local filesystem. If that's the intention, this comment needs to be updated.